src/OpenFeature.DependencyInjection/OpenFeatureServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,53 @@
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using OpenFeature.Internal;
+using OpenFeature.Model;
+
+namespace OpenFeature;
+
+/// <summary>
+/// Contains extension methods for the <see cref="IServiceCollection"/> class.
+/// </summary>
+public static class OpenFeatureServiceCollectionExtensions
+{
+    /// <summary>
+    /// This method is used to add OpenFeature to the service collection.
+    /// OpenFeature will be registered as a singleton.
+    /// </summary>
+    /// <param name="services"><see cref="IServiceCollection"/></param>
+    /// <param name="configure">the desired configuration</param>
+    /// <returns>the current <see cref="IServiceCollection"/> instance</returns>
+    public static IServiceCollection AddOpenFeature(this IServiceCollection services, Action<OpenFeatureBuilder> configure)
+    {
+        Guard.ThrowIfNull(services);
+        Guard.ThrowIfNull(configure);
+
+        services.TryAddSingleton(Api.Instance);
+        services.TryAddSingleton<IFeatureLifecycleManager, FeatureLifecycleManager>();
+
```

Comment:
Instead of having branching in all places where we would like to consume `ExecutionContext` we could always register empty instance of `ExecutionContext`. This will end up with more "pure" methods.
`services.TryAddSingleton(ExecutionContext.Empty);`

Reply:
We don’t register the EvaluationContext as a singleton because the client can operate without any context or with null. Under the hood, it checks and sets the Empty context as needed. The purpose of IsContextConfigured is to avoid redundant service resolution and nullability checks. If no context is registered, we bypass fetching the EvaluationContext entirely, preventing unnecessary GetService or GetRequiredService calls.

Also, when the context is registered, I register it as transient. The reason for this is to control its lifecycle within the scope of the class where it’s used—in this case, IFeatureClient. Will be confusing to register the context as transient in one case and Empty as singleton in another. However, this approach ensures proper lifecycle management based on the context's availability and usage.

Additionally, the IsContextConfigured check is only performed once during the service provider building process. In scenarios where context resolution isn’t required, the IFeatureClient can still function without a context, but engineers can always explicitly use ExecutionContext.Empty if needed